rust-doc-style-guide

```diff
@@ -0,0 +1,268 @@
+  - [Appenndix A Existing Style Guides](#appenndix-a-existing-style-guides)
+    - [Documentation](#documentation)
+    - [Code](#code)
+
+
+## Heading Titles
+
+The books on Rust usually have the heading titles based on nouns or gerunds:
+
+> #### Design Patterns
+> #### Using Structs to Structure Related Data
+
+### Capitalization
+
+In heading titles, capitalize the first letter of every word **except for**:
+
+- Articles (a, an, the); unless an article is the first word.
+
+  > #### Defining an Enum
+  > #### A First Attempt at Rust
+
+- Coordinating conjunctions (and, but, for, or, nor).
+
+  > #### Packages and Crates
+
+- Prepositions of four letters or less; unless these prepositions are the first or last words.
+  - Prepositions of _five_ letters and above should be capitalized (Before, Through, Versus, Among, Under, Between, Without, etc.).
+
+  > #### Peripherals as State Machines
+
+Do not capitalize names of functions, commands, packages, websites, etc.
+
+> #### What is `rustc`
+> #### Publishing on crates.io
+
+See also, the [Monospace and Other Formatting](#monospace-and-other-formatting) section.
+
+In hyphenated words, do not capitalize the parts following the hyphens.
+
+> #### Built-in Targets
+> #### Allowed-by-default Lints
+
+
+## Linking
+
+### Adding Links
+
+The books on Rust usually use [link variables][stackoverflow-link-var] defined right after the paragraph where they are used. For variables, use names that give a clue on where the link leads. It will simplify link maintenance.
+
+[stackoverflow-link-var]: https://stackoverflow.com/a/27784490/10308406
+
+> [`espup`][espup-github] is a tool that simplifies installing and maintaining the components required to develop Rust applications.
+
+> [espup-github]: https://github.com/esp-rs/espup
+
+### Formatting
+
+The books on Rust usually use the following link formatting:
+
+> As mentioned in the [Environment Variables](https://rust-lang.github.io/rustup/installation/index.html) section, ...
+
+> For more details, see the [Windows](rustup-book-windows) chapter in The rustup book.
+
+[book-ja](https://github.com/rust-lang-ja/book-ja/blob/master-ja/style-guide.md) also suggests:
+
+- Make intra-book links relative, so they work both online and locally
+
+Do NOT turn long phrases into links
+
+  > ❌ See the [Rust Reference’s section on constant evaluation](https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html) for more information on what operations can be used when declaring constants.
+
+Also take into account the following
+
+- Do not provide a link to the same location repeatedly in the same or adjacent paragraphs without a good reason, especially using different link text.
+- Do not use the same link text to refer to different locations.
+
+  > `espup` might have a section in a book and a github repo. In this case, see the [`espup`](#espup) section and [`espup` repo](https://hackmd.io/-cf0jVTqRqm9O7GM5JgLiA?both).
```

Comment:
Would it be better to use the link text [`espup` repo](https://hackmd.io/-cf0jVTqRqm9O7GM5JgLiA?both) or [esp-rs/espup](https://hackmd.io/-cf0jVTqRqm9O7GM5JgLiA?both) repo?